Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -338,12 +338,6 @@ WebProcessProxy::~WebProcessProxy()
 
     WebPasteboardProxy::singleton().removeWebProcessProxy(*this);
 
-#if HAVE(DISPLAY_LINK)
-    // Prewarmed / cached processes may not have a process pool on destruction.
-    if (RefPtr processPool = m_processPool.get())
```

Comment:
Personally, I'd feel better if we replaced this with an assertion that checks that we are no longer registered as a client for any existing DisplayLinks.

Alternatively, we can simply fix the recent regression and not ref the process pool:
```
// Unable to ref the process pool as it may have started destruction.
if (auto* processPool = m_processPool.get())
    processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
```

As it stands, the currently PR looks a little risky.

Reply:
Yes, I think ideally we fix the crash first and foremost. It's already Friday night here so it would be awesome if you guys can do something about it 😇

Reply:
To check that we would need to access the process pool. Which leads us to the same problem. That's why I've added an assertion to DisplayLink destructor that it has no associated clients instead. 

This fix will also work:
```
// Unable to ref the process pool as it may have started destruction.
if (auto* processPool = m_processPool.get())
    processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
```

Reply:
If you upload the fix I suggested:
```
// Unable to ref the process pool as it may have started destruction.
if (auto* processPool = m_processPool.get())
    processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
```
Then I'd be happy to r+ it. It would be a safe / targeted fix for a recent regression.

Reply:
@cdumez https://github.com/WebKit/WebKit/pull/19971

Reply:
> Alternatively, we can simply fix the recent regression and not ref the process pool:
> 
> ```
> // Unable to ref the process pool as it may have started destruction.
> if (auto* processPool = m_processPool.get())
>     processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
> ```
> 
> As it stands, the currently PR looks a little risky.

Yes, but dereferencing a pointer to an object that is in the middle of destruction also looks risky to me ))

Reply:
That's not new behavior though, and actually not that uncommon in WebKit.